src/util/Util.js

```diff
@@ -267,6 +267,21 @@ class Util {
     return { animated: Boolean(m[1]), name: m[2], id: m[3] || null };
   }
 
+  /**
+   * Resolves a partial emoji object from an {@link EmojiIdentifierResolvable}, without checking a Client.
+   * @param {EmojiIdentifierResolvable} emoji Emoji identifier to resolve
+   * @returns {Object} Object with possible `animated`, `name`, and `id` properties
+   * @private
+   */
+  static resolvePartialEmoji(emoji) {
+    if (typeof emoji === 'undefined') return null;
+    if (typeof emoji === 'string') {
+      return emoji.length ? (/^\d{17,19}$/.test(emoji) ? { id: emoji } : Util.parseEmoji(emoji)) : null;
+    }
+    const { id, name, animated } = emoji;
+    return { id, name, animated };
```

Comment:
Kinda redundant, no?

Reply:
Somewhat - this was done to match the typing of what I've said it returns, and what will be stored in the #emoji prop of components.

Reply:
I don't think its redundant, it also prevents returning non emoji props. However, I might suggest throwing an error if both id and name are undefined in between these lines, as in that case you do not have a valid emoji.